governance/program/src/instruction.rs

```diff
@@ -0,0 +1,264 @@
+    },
+
+    /// [Requires Admin token]
+    /// Delete Proposal entirely.
+    ///
+    ///   0. `[writable]` Proposal state account pub key.
+    ///   1. `[writable]` Admin account
+    ///   2. `[writable]` Admin Validation account.
+    ///   3. `[]` Proposal account pub key.
+    ///   4. `[]` Transfer authority.
+    ///   5. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   6. `[]` Token program account.
+    DeleteProposal,
+
+    /// [Requires Signatory token]
+    /// Burns signatory token, indicating you approve of moving this Proposal from Draft state to Voting state.
+    /// The last Signatory token to be burned moves the state to Voting.
+    ///
+    ///   0. `[writable]` Proposal State account
+    ///   1. `[writable]` Signatory account
+    ///   2. `[writable]` Signatory Mint account
+    ///   3. `[]` Proposal account
+    ///   4. `[]` Transfer authority
+    ///   5. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   6. `[]` Token program account.
+    ///   7. `[]` Clock sysvar.
+    SignProposal,
+
+    /// [Requires Voting tokens]
+    /// Burns voting tokens, indicating you approve and/or disapprove of running this set of transactions. If you tip the consensus,
+    /// then the transactions can begin to be run at their time slots when people click execute. You are then given yes and/or no tokens
+    ///
+    ///   0. `[writable]` Governance voting record account
+    ///                   Can be uninitialized or initialized(if already used once in this proposal)
+    ///                   Must have address with PDA having seed tuple [Governance acct key, proposal key, your voting account key]
+    ///   1. `[writable]` Proposal State account
+    ///   2. `[writable]` Your Vote account.
+    ///   3. `[writable]` Your Yes Vote account
+    ///   4. `[writable]` Your No Vote account
+    ///   5. `[writable]` Vote Mint account
+    ///   6. `[writable]` Yes Vote Mint account
+    ///   7. `[writable]` No Vote Mint account
+    ///   8. `[]` Source Token Mint account
+    ///   9. `[]` Proposal account
+    ///   10. `[]` Governance account
+    ///   12. `[]` Transfer authority
+    ///   13. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   14. `[]` Token program account
+    ///   15. `[]` Clock sysvar
+    Vote {
+        /// Yes/No  with amount of votes
+        vote: Vote,
+    },
+
+    /// Executes a command in the Proposal
+    ///
+    ///   0. `[writable]` Transaction account you wish to execute
+    ///   1. `[writable]` Proposal State account
+    ///   2. `[]` Program being invoked account
+    ///   3. `[]` Proposal account
+    ///   4. `[]` Governance account
+    ///   5. `[]` Governance program account pub key
+    ///   6. `[]` Clock sysvar
+    ///   7+ Any extra accounts that are part of the instruction, in order
+    Execute,
+
+    /// Creates an empty governance vote record
```

Comment:
I don't think we'll need this instruction unless we run into a compute limit.  Otherwise, `Vote` can do this work